### PR TITLE
Brownbag

### PR DESCRIPTION
**Pull request by shadone**
Commit of shame. Previous pull requests were not entirely correct, I believe this is a more conformant version

### CODE REVIEW

**Comment by shadone** [comment]
Wait, forgot to test.

File: index.js [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
+      results.push(element);
+    }
+  });
+  return results;
+}
+
 function generate(tuples, options) {
+  if (tuples.length === 0) {
+    // according to rfc1341 there should be at least one encapsulation
+    return undefined;
```
Comment (sergi) [inline index.js]: I would just throw an `Error`

File: index.js [hunk 2]
```diff
@@ -1,37 +1,74 @@
...
+
   options = options || {};
+  var from = options.from;
+  var preamble = options.preamble;
+  var epilogue = options.epilogue;
   var boundary = options.boundary || uuid();
+
+  if (boundary.length > 70) {
+    // rfc defines boundary to be between 1 and 70 characters long.
+    return undefined;
```
Comment (sergi) [inline index.js]: Again, throw `Error`

File: index.js [hunk 3]
```diff
@@ -1,37 +1,74 @@
...
       'Content-Disposition: attachment; filename="' + filename + '"',
-    ].join(CRLF) + CRLF;
+    ];
 
-    return [delimiter, fileHeaders, tuple.content].join(CRLF);
+    var bodyPart = headers.join(CRLF) + CRLF + CRLF + tuple.content;
+
+    var encapsulation = delimiter + CRLF + bodyPart;
+
+    return encapsulation;
```
Comment (sergi) [inline index.js]: Let's just return `delimiter + CRLF + bodyPart`. No need for `encapsulation variable.

File: index.js [hunk 4]
```diff
@@ -1,37 +1,74 @@
...
+    return typeof element !== 'undefined' && element !== null;
+  }
+
+  array.forEach(function(element) {
+    if (truthy(element)) {
+      results.push(element);
+    }
+  });
+  return results;
+}
```
Comment (sergi) [inline index.js]: This whole function could be simply

``` javascript
function compact(array) {
  return array.filter(function truthy(element) {
    return typeof element !== 'undefined' && element !== null;
  });
}
```

Comment (sergi) [inline index.js]: Also, `compact` would now include empty strings. Is that your intention?

Comment (shadone) [inline index.js]: hm, probably not. What's the best way to filter our empty strings?

```
element !== ''
```

?

File: index.js (hunk 1 above)
Comment (shadone) [inline index.js]: done

File: index.js (hunk 2 above)
Comment (shadone) [inline index.js]: done

File: index.js (hunk 3 above)
Comment (shadone) [inline index.js]: done

File: index.js [hunk 5]
```diff
@@ -1,37 +1,62 @@
 var uuid = require('node-uuid');
 var CRLF = '\r\n';
 
+function compact(array) {
+  return array.filter(function(element) {
+    return typeof element !== 'undefined' && element !== null;
```
Comment (sergi) [inline index.js]: To also filter empty strings I think that you could simply use

``` javascript
return array.filter(function(element) { return !!element; }) 
```

At which point you could simply put this below, where you use `compact`:

``` javascript
var multipartBody = [
    preamble ? CRLF + preamble : undefined,
    encapsulations.join(''),
    closeDelimiter,
    epilogue ? CRLF + epilogue : undefined,
 ].filter(function(element) { return !!element; });
```

...and get rid of `compact` althogether.

Comment (shadone) [inline index.js]: done.

**Comment by sergi** [comment]
Thanks!

**Pull request merged**